src/D2L.Security.OAuth2/Keys/UnsignedToken.cs

```diff
@@ -11,7 +9,7 @@ namespace D2L.Security.OAuth2.Keys {
 	public sealed class UnsignedToken {
 		private readonly string m_issuer;
 		private readonly string m_audience;
-		private readonly ReadOnlyCollection<Claim> m_claims;

```

Comment:
So I take it we were misusing the `Claim` type (from System.Security.Claims, not the JWT stuff) here?

Reply:
Unsure, the `Claim` type was valid for use with the `JwtSecurityToken`, but Microsoft's implementation was strict around not allowing claims to be nested or contain arrays.

Reply:
At least not through manual construction.. It supported a BinaryReader which could have in theory allowed for anything.